cpmpy/expressions/core.py

```diff
@@ -110,6 +111,64 @@ class Expression(object):
     - any ``__op__`` python operator overloading
     """
 
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        """
+        Handle NumPy ufuncs for CPMpy expressions.
+        
+        This method is called when a NumPy ufunc is applied to a CPMpy expression.
+        It ensures that CPMpy expressions handle the operation correctly.
+        """
+        # Only handle the __call__ method
+        if method != "__call__":
+            return NotImplemented
+        
+        # Map NumPy ufuncs to CPMpy method names:
+        # ufunc, operator, reverse operator (when needed, as expressions are in rhs)
+        op_map = {
+            # Comparisons
+            np.equal: [Expression.__eq__, Expression.__eq__],
+            np.not_equal: [Expression.__ne__, Expression.__ne__],
+            np.less: [Expression.__lt__, Expression.__gt__],
+            np.less_equal: [Expression.__le__, Expression.__ge__],
+            np.greater: [Expression.__gt__, Expression.__lt__],
+            np.greater_equal: [Expression.__ge__, Expression.__le__],
+            # Arithmetic
+            np.add: [Expression.__add__, Expression.__radd__],
+            np.subtract: [Expression.__sub__, Expression.__rsub__],
+            np.multiply: [Expression.__mul__, Expression.__rmul__],
+            np.floor_divide: [Expression.__floordiv__, Expression.__rfloordiv__],
+            np.true_divide: [Expression.__truediv__, Expression.__rtruediv__],  # Will warn and use floordiv
+            np.mod: [Expression.__mod__, Expression.__rmod__],
+            np.power: [Expression.__pow__, Expression.__rpow__],
+            # Logical
+            np.logical_and: [Expression.__and__, Expression.__rand__],
+            np.logical_or: [Expression.__or__, Expression.__ror__],
+            np.logical_xor: [Expression.__xor__, Expression.__rxor__]
+        }
+        
+        # Handle binary operations
+        if len(inputs) == 2 and ufunc in op_map:
+            x, y = inputs
+
+            if isinstance(x, Expression):
```

Comment:
wait, Re: Ignace' comment... that is this case (in case we would translate all expression operators to ufuncs, which is what numpy recommends it seems), then we could check here if broadcasting needs to happen?
that would be a major overhaul, but it could lead to consistent broadcasting instead of only NDVarArray being broadcast-aware...